File: src/main/kotlin/shop/carpetandbeer/model/Item.kt
package shop.carpetandbeer.model

import lombok.Data
import org.jetbrains.annotations.NotNull
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.net.URL

enum class Category {
    BEER, CARPET
}

@Document(collection = "items")
@Data
class Item(
    @Id
    var id: String?,
    @NotNull
    var name: String?,
    @NotNull
    var price: Double?,
    @NotNull
    var description: String?,
    @NotNull
    var category: Category?,
    var imgURL: String?
)

class ItemRequest {
    var name: String? = null
    var price: Double? = null
    var description: String? = null
    var category: Category? = null
    var imgUrl: String? = null
}

File: src/app/lib/model/Item.dart
class Item {
  final String id;
  final String name;
  final String description;
  final double price;
  final String imageUrl;
  final String category;

  Item(
      {required this.category,
      required this.id,
      required this.name,
      required this.description,
      required this.price,
      required this.imageUrl});

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: json['price'],
      imageUrl: json['imgUrl'],
      category: '',
    );
  }

  factory Item.fromJsonShared(Map<String, dynamic> json) {
    return Item(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: json['price'],
      imageUrl: json['imageUrl'],
      category: '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'imageUrl': imageUrl,
    };
  }
}
